### ISSUE TITLE
"has_key?" deprecated?

### ISSUE THREAD

**Issue by robertgj** [inukshuk/jekyll-scholar#193]
I am new to ruby and jekyll and am trying to build an example post with a bibliography.

I had the following error message:
```
 Liquid Exception: undefined method `has_key?' for #<Jekyll::Drops::DocumentDrop:0x0055c1142bc228> in /home/robj/CHANGES/rjweb.C013/jekyll/example/_posts/2017-06-25-welcome-to-jekyll.markdown
/home/robj/.gem/ruby/gems/jekyll-scholar-5.9.1/lib/jekyll/scholar/tags/bibliography.rb:19:in `render': undefined method `has_key?' for #<Jekyll::Drops::DocumentDrop:0x0055c1142bc228> (NoMethodError)
```
I made the following change to get bibliographies to work:

```--- jekyll-scholar-master/lib/jekyll/scholar/tags/bibliography.rb	2017-05-04 23:04:24.000000000 +1000
+++ /home/robj/.gem/ruby/gems/jekyll-scholar-5.9.1/lib/jekyll/scholar/tags/bibliography.rb	2017-06-25 23:37:42.521358150 +1000
@@ -16,7 +16,7 @@
         set_context_to context
 
         # Add bibtex files to dependency tree
-        if context.registers[:page] and context.registers[:page].has_key? "path"
+        if context.registers[:page] and context.registers[:page].key? "path"
           bibtex_paths.each do |bibtex_path|
             site.regenerator.add_dependency(
               site.in_source_dir(context.registers[:page]["path"]),

**Comment by IgnoredAmbience** [comment]
Which version of Jekyll are you using this with? It potentially looks like an incompatibility with a new version, as the code that breaks is 2 years old.

**Comment by IgnoredAmbience** [comment]
Updated to Jekyll 3.5, and the tests are still passing... Continuing to attempt to reproduce.

**Comment by IgnoredAmbience** [comment]
Reproduced successfully. Have confirmed the fix works. Still unsure why tests are not failing as expected without the fix.

**Comment by robertgj** [comment]
Thanks for your work on Jekyll-scholar

Output from "gem info -d" gives:

jekyll (3.5.0, 3.4.3)
    Author: Tom Preston-Werner
    Homepage: https://github.com/jekyll/jekyll
    License: MIT
    Installed at (3.5.0): /home/robj/.gem/ruby
                 (3.4.3): /home/robj/.gem/ruby

    A simple, blog aware, static site generator.

**Comment by IgnoredAmbience** [comment]
This has now been fixed in v5.10.1.
